src/applications/discharge-wizard/utilities/page-navigation.js

```diff
@@ -93,31 +93,53 @@ export const navigateForward = (
           !formResponses[nextShortName] &&
           editMode
         ) {
-          updateRouteMap([...routeMap, ROUTES?.[nextShortName]]);
+          if (routeMap[routeMap.length - 1] !== ROUTES?.[nextShortName]) {
+            if (valueHasChanged) {
+              const index = routeMap.indexOf(ROUTES[SHORT_NAME]);
+              const newRouteMap = routeMap.slice(0, index + 1);
+              setRouteMap([...newRouteMap, ROUTES?.[nextShortName]]);
+            } else {
+              setRouteMap([...routeMap, ROUTES?.[nextShortName]]);
+            }
+          }
+
           pushToRoute(nextShortName, router);
           return;
         }
 
+        // This handles the edge case of the edit question not being skipped
```

Comment:
Consider: don't be afraid of longer variable names or more detailed comments to explain complex behavior. For instance, "edit question" may not be clear to a stranger to this code; maybe "question being edited." (Same with variable `editQuestion`; this could be `questionSelectedToEdit`) Not sure if my suggestions make it more clear; just offering some perspective

Reply:
That makes sense I can make these changes